src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,313 @@
+        return self._addinbase
+
+    @property
+    def application(self):
+        return self._application
+
+    @property
+    def experiment(self):
+        return self._experiment
+
+    def __del__(self):
+        """
+        Uses Python garbage collection method used to terminate
+        AddInProcess.exe, if this is not done, LightField will not reopen
+        properly.
+        """
+        self.automation.Dispose()
+        logger.info('Closed AddInProcess.exe')
+
+    def set(self, setting, value):
+        """
+        Helper function for setting experiment parameters with basic value
+        checking.
+        """
+        if self.experiment.Exists(setting):
+            if self.experiment.IsValid(setting, value):
+                self.experiment.SetValue(setting, value)
+            else:
+                # TODO: might need to add a proper exception
+                logger.error(f'Invalid value: {value} for setting: {setting}.')
+        else:
+            # TODO: might need to add a proper exception
+            logger.error(f'Invalid setting: {setting}.')
+
+    def get(self, setting):
+        """
+        Helper function for getting experiment parameters with a basic check
+        to see if a specific setting is valid.
+        """
+        if self.experiment.Exists(setting):
+            value = self.experiment.GetValue(setting)
+        else:
+            value = []
+            logger.error(f'Invalid setting: {setting}.')
+        return value
+
+    def load_experiment(self, value):
+        self.experiment.Load(value)
+
+    def acquire(self):
+        """
+        This function retrieves image data with special consideration for the unique configuration of the camera,
+        which possesses a single vertical pixel. The following describes the data handling:
+
+        - **Single Frame**:
+        - Data is returned as a 2D array representing raw counts from each pixel.
+        - A vertical section (spanning 400 pixels) represents a range for wavelength averaging.
+
+        - **Multiple Frames**:
+        - Data from successive exposures is added as a new dimension, resulting in a 3D array.
+        - Averaging across frames can be done by summing over this additional dimension.
+        """
+
+        acquisition_time = self.get(lf.AddIns.CameraSettings.ShutterTimingExposureTime) #to allow the camera to capture the desired amount of data during the specified exposure time.
+        num_frames = self.get(lf.AddIns.ExperimentSettings.FrameSettingsFramesToStore)
+        self.experiment.Acquire()
+
+        sleep(0.001 * acquisition_time * num_frames)  #sleep delay that waits for the exposure duration of the camera
+
+        while self.experiment.IsRunning:
+            sleep(0.1)  #loop that repeatedly checks whether the experiment is still running so it decided when to move to next block of code
+
+        last_file = self.application.FileManager.GetRecentlyAcquiredFileNames().get_Item(0)
```

Comment:
This is a very good question. I am going to leave this as unresolved and try work on a fix

Reply:
I have come across possible way to handle this, not sure which one will work. 

The first option is to set both of these before acquiring. If the time we attach goes as far down as milliseconds then it should solve our problem.

``` 
experiment.set(ExperimentSettings.FileNameGenerationAttachDate, true)
experiment.set(ExperimentSettings.FileNameGenerationAttachTime, true)
```

**OR**

We could try this method, according to the docs in the .chm file this will attach a numeric increment to the file name

``` 
experiment.set(ExperimentSettings.FileNameGenerationAttachIncrement, true)
```

Reply:
Edit: The first option wont go past seconds so its no use to us if we are taking multiple a second also I think its already enabled by default so its useless to us